engine/access/rest/request_decorator.go

```diff
@@ -0,0 +1,51 @@
+package rest
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+
+	"github.com/onflow/flow-go/engine/access/rest/middleware"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// a convenience wrapper around the http request to make it easy to read request params
+type requestDecorator struct {
+	*http.Request
+	expandFields map[string]bool
+	selectFields map[string]bool
+}
+
+func newRequestDecorator(r *http.Request) *requestDecorator {
+	decoratedReq := &requestDecorator{
+		Request: r,
+	}
+	decoratedReq.expandFields, _ = middleware.GetFieldsToExpand(r)
```

Comment:
Are these even middlewares anymore? Do you think it would make sense for those functions to be moved in this decorator?